tests/python_functional/functional_tests/conftest.py

```diff
@@ -74,7 +74,11 @@ class PortAllocator():
     CURRENT_DYNAMIC_PORT = 30000
 
 
-@pytest.fixture(scope="module")
-def some_port():
+def _get_next_port():
     PortAllocator.CURRENT_DYNAMIC_PORT += 1
     return PortAllocator.CURRENT_DYNAMIC_PORT
+
+
+@pytest.fixture(scope="session")
```

Comment:
Do we need fixtures in the end? If `_get_next_port` would be public, we could just call it.

Reply:
I think the public function will work in this situation. @mitzkia ?

(My approach was from visibility. With scope="session" we guarantee the lifetime of the inside Class in the pytest way.)

Reply:
Without the fixture we should have instantiated the class first which would be very inconvenient. I don't like that way.
Or we don't need this whole class at all. It could be a simple (I mean not a class member) function.

Reply:
In the current code state if we remove this fixture (port_allocator) we should not instantiate the class (if you mean PortAllocator).
The difference should be something like this on a caller side:

```diff
+from functional_tests.conftest import get_next_port
...
-@pytest.fixture
-def snmptrapd(port_allocator):
-    server = SNMPtrapd(port_allocator())
+@pytest.fixture
+def snmptrapd():
+    server = SNMPtrapd(get_next_port())
```

From my point of view it is almost equivalent change.
From one side fixtures gives us a little flexibility about their life-cycles and usage (do not require import) , but in the other hand they (fixtures) can be very quickly multiplies, and "everything" will be a fixture, which is not so good. 
For now I would say we can keep this fixture, but later we should think them over (what can be a fixture and what not).